ECH in enterprise networks (#29)
thomas-fossati/draft-tls13-iot: draft-ietf-uta-tls13-iot-profile.md

# Server Name Indication

This specification mandates the implementation of the Server Name Indication (SNI)
extension. Where privacy requirements require it, the ECH (Encrypted Client Hello)
extension {{?I-D.ietf-tls-esni}} prevents an on-path attacker to determine the domain
name the client is trying to connect to.

> Note however that in environments where protection against malware is essential (e.g., enterprise networks, industrial IoT deployments), using ECH may not be an option because network administrators want to be able to filter the DNS traffic generated by IoT devices in order to prevent access to malicious content.

Author: ```suggestion
Note however that in environments where protection against malware is essential (e.g., enterprise networks, industrial IoT deployments), using ECH may not be an option because network administrators want to be able to filter the DNS traffic generated by IoT devices in order to prevent access to potentially malicious content.
```

# Server Name Indication

This specification mandates the implementation of the Server Name Indication (SNI)
extension. Where privacy requirements require it, the ECH (Encrypted Client Hello)
extension {{?I-D.ietf-tls-esni}} prevents an on-path attacker to determine the domain
name the client is trying to connect to.

Since the Encrypted Client Hello extension requires use of Hybrid Public Key
Encryption (HPKE) {{?I-D.irtf-cfrg-hpke}} and additional protocols require
further protocol exchanges and cryptographic operations, there is a certain
overhead associated with this privacy feature.

> Note that in industrial IoT deployments the use of ECH may not be an option because network administrators inspect DNS traffic generated by IoT devices in order to detect malicious behaviour

Reviewer: ```suggestion
Note that in industrial IoT deployments the use of ECH may not be an option because network administrators inspect DNS traffic generated by IoT devices in order to detect malicious behaviour.
```

> Note that in environments where protection against malware is essential (e.g., enterprise networks, industrial IoT deployments), using ECH may not be an option because network administrators want to be able to filter the DNS traffic generated by IoT devices in order to prevent access to potentially malicious content.

Reviewer: ```suggestion
Note that in industrial IoT deployments the use of ECH may not be an option because network administrators inspect DNS traffic generated by IoT devices in order to detect malicious behaviour
```
